src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -247,11 +247,17 @@ impl Validator<'mir, 'tcx> {
             return;
         }
 
-        // If an operation is supported in miri (and is not already controlled by a feature gate) it
+        // If an operation is supported in miri it
         // can be turned on with `-Zunleash-the-miri-inside-of-you`.
-        let is_unleashable = O::IS_SUPPORTED_IN_MIRI && O::feature_gate().is_none();
+        let is_unleashable = O::IS_SUPPORTED_IN_MIRI;
 
         if is_unleashable && self.tcx.sess.opts.debugging_opts.unleash_the_miri_inside_of_you {
+            // If all skipped operations could also be turned on by a feature gate, then we want
+            // to emit an error. So when we see a non-feature gate operation being skipped, disable
+            // that error reporting.
+            if O::feature_gate().is_none() {
```

Comment:
I think the point of the flag is to make the static checker shut up. So, whether or not something requires a feature gate in the static checker should not have any influence on what happens.

So, I am not a big fan of this proposal.